Detect Fuse on Karaf

noahl opened the pull request:
Scan running processes and use the locate command to detect JBoss Fuse
running on Apache Karaf.

As a bonus, also add a fact 'jboss.fuse.init-files' which looks for
init files that contain the string 'fuse'. It has some false
positives, but could be useful for finding systems that probably have
fuse.

Closes #480 . Closes #372 .

Commit:
Detect Fuse on Karaf

Scan running processes and use the locate command to detect JBoss Fuse
running on Apache Karaf.

As a bonus, also add a fact 'jboss.fuse.init-files' which looks for
init files that contain the string 'fuse'. It has some false
positives, but could be useful for finding systems that probably have
fuse.

Commit:
Fix typo in test

chambridge commented on rho_playbook.yml:
```diff
@@ -16,6 +16,7 @@
     - dmi
     - etc_release
     - file_contents
+    - fuse_on_karaf
```
I would prefer we keep the jboss work together and call the role `jboss_fuse_on_karaf` placing it towards the end with the `jboss_eap` role

noahl replied:
I moved everything to the `jboss_eap` role and updated the fact_docs and `remote_programs.rst`. Thank you!

chambridge reviewed (commented):
Comment on role name, although its more about grouping with the other middleware roles.

You need to generate the latest facts doc.

Also I think updates are likely needed to:
https://github.com/quipucords/rho/blob/master/doc/source/remote_programs.rst

Commit:
Update in response to review comments

- Move Fuse-on-Karaf to jboss_eap Ansible role
- Update fact docs
- Update remote_programs.rst

Thanks to chambrid for the review.

chambridge commented on rho_playbook.yml:
```diff
@@ -16,7 +16,6 @@
     - dmi
     - etc_release
     - file_contents
-    - fuse_on_karaf
```
Didn't mean for you to remove the role. Just rename it and have it located with `jboss_eap` in the playbook.
Example
```
- jboss_eap
- jboss_eap_scan
- brms
- fuse
- jboss_fuse_on_karaf
```

chambridge commented on roles/jboss_eap/tasks/main.yml:
```diff
@@ -115,3 +115,88 @@
       become: true
       when: '"jboss.eap.init-files" in facts_to_collect'
+
+# Tasks that can locate a KARAF_HOME directory
```
Sorry for the confusion, just wanted the role to be run in a different order. Not combined with the eap role.

noahl replied:
My mistake! I just pushed another change which I think is what you are looking for.

Commit:
Move Fuse-on-Karaf tasks to their own Ansible role

Thanks to chambrid for the review.
